backend/main/chapters/c09_combining_booleans.py

```diff
@@ -0,0 +1,638 @@
+        """
+        program_in_text = False
+
+        def program(self):
+            def is_friend(name):
+                if name == "Alice" or name == "Bob":
+                    return True
+                else:
+                    return False
+
+            assert_equal(is_friend("Alice"), True)
+            assert_equal(is_friend("Bob"), True)
+            assert_equal(is_friend("Charlie"), False)
+
+    class FurtherImprovement(VerbatimStep):
+        """
+We can do even better. Notice that
+
+    name == "Alice" or name == "Bob"
+
+is a boolean, and both `return` statements are returning booleans `True` or `False`.
+The function returns `True` when `name == "Alice" or name == "Bob"` is `True`,
+and it returns `False` when `name == "Alice" or name == "Bob"` is `False`.
+So we could simply return the boolean `name == "Alice" or name == "Bob"` itself in both cases!
+
+This is a common pattern for simplifying your code. If you ever find yourself writing code like:
+
+    if x:
+        return True
+    else:
+        return False
+
+where `x` itself is a boolean, you can always simplify this block of code to:
+
+    return x
+
+Now in the editor, make the appropriate simplification as explained above, and run your code again.
+        """
+
+        program_in_text = False
+
+        def program(self):
+            def is_friend(name):
+                return name == "Alice" or name == "Bob"
+
+            assert_equal(is_friend("Alice"), True)
+            assert_equal(is_friend("Bob"), True)
+            assert_equal(is_friend("Charlie"), False)
+
+    class ACommonMistake(VerbatimStep):
+        """
+Take careful note of how we wrote the condition:
+
+    return name == "Alice" or name == "Bob"
+
+A common mistake is to write this instead:
+
+    return name == "Alice" or "Bob"
+
+It makes sense if you read it like English: "If `name` is equal to either "Alice" or "Bob", then return True...".
+But Python is not English, and that's not how `or` works.
+
+Replace the `return` line in the code with the above line, and try running it again.
+        """
+        program_in_text = False
+
+        def program(self):
+            def is_friend(name):
+                return name == "Alice" or "Bob"
+
+            assert_equal(is_friend("Alice"), True)
+            assert_equal(is_friend("Bob"), True)
+            assert_equal(is_friend("Charlie"), False)
+
+    class InspectWithBirdseye(ACommonMistake):
+        text = """
+The second and third tests fail! Our function seems to be doing the wrong thing:
+it returns `Bob` (a string, not a boolean!) when `name` is `Charlie`. What is going on?
+Try inspecting the code with Birdseye. Inspect the `return` statements of each `is_friend` calls carefully.
+(Use the blue arrow buttons)
+        """
+
+        def check(self):
+            return super().check() and self.code_source == "birdseye"
+
+    class AnExercise(ExerciseStep):
+        """
+When we inspect it with Birdseye, we can see that:
```

Comment:
```suggestion
When we inspect it with Bird's Eye, we can see that:
```

I know I wrote this part, but I've since decided to spell it like that within futurecoder. I feel weird about it but the one word spelling is good for `import birdseye` and `pip install birdseye`, not this.

Reply:
Got it.